### ISSUE TITLE
handle ssl encryption in route

### ISSUE THREAD

**Issue by Ctaque** [tightenco/ziggy#109]
I'm currently deploying to heroku and it appears that I get this error:

> Mixed Content: The page at 'https://onsortirapeutetre.herokuapp.com/' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://onsortirapeutetre.herokuapp.com/api/city'. This request has been blocked; the content must be served over HTTPS.

Does it handle ssl encryption? Because when i do 
`route('api....').url()`
this is the result : 
`http://onsortirapeutetre.herokuapp.com/api/excursion`

**Comment by Ctaque** [comment]
I looked into this is noticed that the method prepareDomain() uses the url method of laravel, which as described in the documentation creates a url designed for non ssl encryption in contrary to the secure_url() method.

I think that the environment variable APP_URL could be used and parsed to create the domain base variables.

```
private function prepareDomain()
    {
        $url = url('/');
        $parsedUrl = parse_url($url);
        $this->baseUrl = $url . '/';
        $this->baseProtocol = array_key_exists('scheme', $parsedUrl) ? $parsedUrl['scheme'] : 'http';
        $this->baseDomain = array_key_exists('host', $parsedUrl) ? $parsedUrl['host'] : '';
        $this->basePort = array_key_exists('port', $parsedUrl) ? $parsedUrl['port'] : 'false';
    }
```

This is what i think about : 
Just add : $url = env('APP_URL', url('/'));
```
private function prepareDomain()
    {
        $url = env('APP_URL', url('/'));
        $parsedUrl = parse_url($url);
        $this->baseUrl = $url . '/';
        $this->baseProtocol = array_key_exists('scheme', $parsedUrl) ? $parsedUrl['scheme'] : 'http';
        $this->baseDomain = array_key_exists('host', $parsedUrl) ? $parsedUrl['host'] : '';
        $this->basePort = array_key_exists('port', $parsedUrl) ? $parsedUrl['port'] : 'false';
    }
```

**Comment by Ctaque** [comment]
What do you think of this @DanielCoulbourne  ?

**Comment by DanielCoulbourne** [comment]
Fixed by #111
